10_date_timestamps_and_lies.md

```diff
@@ -0,0 +1,191 @@
+# External Data Sources - Die or die trying
+
+### Dates, Timestamps and lies
+
+I was working once with some legacy database on MySQL and one of the most common
+problems is actually dealing with dates and timestamps. But so we think.
+
+It is also not enough to know how to use the `jdbc` connector to know how to
+pull data from MySQL. You actually need to understand the mechanism of the
+connector, the database; in occurence MySQL and also Spark's in these cases.
+
+So here is the drill. I was reading some data from MySQL. Some columns are of
+type `timestamp` with a default value is "0000-00-00 00:00:00". Something like
+this :
+
+```
++--------------------------+---------------+---------------------+----------------+
+| Field                    | Type          | Default             | Extra          |
++--------------------------+---------------+---------------------+----------------+
+| lastUpdate               | timestamp     | 0000-00-00 00:00:00 |                |
++--------------------------+---------------+---------------------+----------------+
+```
+
+Spark doesn't seem to like that :
+
+```
+java.sql.SQLException: Cannot convert value '0000-00-00 00:00:00' from column lastUpdate to TIMESTAMP.
+    at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1055)
+    [...]
+```
+
+***So how to we deal with that ?***
+
+The first thing that comes in mind always when dealing with such issue is converting into
+a string then dealing with the date through parsing with some
+[Joda](http://www.joda.org/joda-time/) or such.
+
+Well, that ~~isn't always~~, is never the best solution.
+
+And here is why :
+
```

Comment:
Does it really make sense to suggest Joda? Spark SQL has pretty much all we need to parse dates.